addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -166,11 +166,28 @@ export class SuggestionService {
                 suggestions.push(partner);
             }
         }
+        const specialSuggestions = this.store.specialMentions
+            .filter((special) => {
+                return (
+                    special.channel_types.includes(thread.channel_type) &&
+                    cleanedSearchTerm.length >= Math.min(4, special.label.length) &&
+                    (special.label.startsWith(cleanedSearchTerm) ||
+                        cleanTerm(special.description.toString()).includes(cleanedSearchTerm))
+                );
+            })
+            .map((special) => ({
+                isSpecial: true,
+                label: special.label,
+                type: special.label,
+                name: special.displayName,
+                description: special.description,
+            }));
```

Comment:
`thread` is an optional param. It is usually provided in channels but not necessarily in chatter, notably in the full composer (that is when you click on the 2 outer arrows to have composer in a dialog).

When attempting mentions there, it crashes because it tries to `thread.channel_type` on `thread` being `undefined`.

Make sure to handle the case where `thread` is `undefined` :)